### ISSUE TITLE
elasticsearch 2.0 handler type for murmur3 not found

### ISSUE THREAD

**Issue by maksutspahi** [elastic/elasticsearch#14828]
I have updated 1.4.2 to 2.0.0. 
I installed plugin 
   `sudo bin/plugin install mapper-murmur3`

I had some data inserted in 1.4.2. After I updated to 2.0.0 I am getting this error while starting elasticsearch.
```
Exception: java.lang.IllegalStateException: unable to upgrade the mappings for the index [abc], reason: [no handler for type [murmur3] declared on field [hash]]
Likely root cause: MapperParsingException[no handler for type [murmur3] declared on field [hash]]
	at org.elasticsearch.index.mapper.core.TypeParsers.parseMultiField(TypeParsers.java:341)
	at org.elasticsearch.index.mapper.core.StringFieldMapper$TypeParser.parse(StringFieldMapper.java:201)
	at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:315)
	at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:228)
	at org.elasticsearch.index.mapper.object.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:137)
	at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:211)
	at org.elasticsearch.index.mapper.DocumentMapperParser.parseCompressed(DocumentMapperParser.java:192)
	at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:368)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:242)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:329)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.upgradeIndexMetaData(MetaDataIndexUpgradeService.java:112)
	at org.elasticsearch.gateway.GatewayMetaState.pre20Upgrade(GatewayMetaState.java:226)
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:85)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
	at <<<guice>>>
	at org.elasticsearch.node.Node.<init>(Node.java:198)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:170)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:270)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
```

My mapping was:
```
PUT /abc
{
  "mappings": {
    "productView": {
      "properties": {
        "itemId": {
           "index": "not_analyzed",
           "type": "string",
           "fields": {
            "hash": {
              "type": "murmur3"
            }
          }
        }
      }
    }
  }
}
```
But when I delete the old data and recreate the index it works without problem.

**Comment by jpountz** [comment]
The issue reproduces for me. It seems to be because we try to upgrade the mappings before plugins have a chance to register mappers. So any mappers that are registered through a plugin are affected by this problem.

**Comment by jpountz** [comment]
@rjernst @imotov Any ideas if/how we can fix it?

**Comment by rjernst** [comment]
I think we need to move registering mappers to the node level (which makes sense anyways).  Instead of getting the mapper service to register them, the mapper service should take in all the custom types as args. I'm not sure if it will directly fix the issue, but at least then it seems like it should be possible.

**Comment by dadoonet** [comment]
> It seems to be because we try to upgrade the mappings before plugins have a chance to register mappers. So any mappers that are registered through a plugin are affected by this problem.

++ That explains as well issues we have seen for mapper attachments plugin. See https://github.com/elastic/elasticsearch-mapper-attachments/issues/180

**Comment by jpountz** [comment]
> I think we need to move registering mappers to the node level

I opened #14896 for this.

**Comment by jpountz** [comment]
This has been fixed by #14896 and then I added tests in #14977. This will be fixed as of 2.1.1. I considered backporting to 2.0 as well but there were so many conflicts that I got a bit afraid of doing more harm than good by introducing new bugs.

**Comment by jmreymond** [comment]
same issue in 2.1.1
root@dedi:/usr/share/elasticsearch# ./bin/plugin list
Installed plugins in /usr/share/elasticsearch/plugins:
    - marvel-agent
    - license
    - mapper-attachments

[2015-12-19 07:31:59,246][INFO ][node                     ] [Manta] version[2.1.1], pid[28723], build[40e2c53/2015-12-15T13:05:55Z]
[2015-12-19 07:31:59,247][INFO ][node                     ] [Manta] initializing ...
[2015-12-19 07:32:00,314][INFO ][plugins                  ] [Manta] loaded [license, marvel-agent, mapper-attachments], sites []
[2015-12-19 07:32:00,356][INFO ][env                      ] [Manta] using [1] data paths, mounts [[/home (/dev/sda4)]], net usable_space [641.6gb], net total_space [857.8gb], spins? [possibly], types [ext4]
[2015-12-19 07:32:04,427][ERROR][gateway                  ] [Manta] failed to read local state, exiting...
java.lang.IllegalStateException: unable to upgrade the mappings for the index [evol2], reason: [Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]]
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:339)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.upgradeIndexMetaData(MetaDataIndexUpgradeService.java:116)
	at org.elasticsearch.gateway.GatewayMetaState.pre20Upgrade(GatewayMetaState.java:228)
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:87)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.elasticsearch.common.inject.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:56)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:86)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.SingleParameterInjector.inject(SingleParameterInjector.java:42)
	at org.elasticsearch.common.inject.SingleParameterInjector.getAll(SingleParameterInjector.java:66)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:85)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.SingleParameterInjector.inject(SingleParameterInjector.java:42)
	at org.elasticsearch.common.inject.SingleParameterInjector.getAll(SingleParameterInjector.java:66)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:85)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:201)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:880)
	at org.elasticsearch.common.inject.InjectorBuilder.loadEagerSingletons(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorBuilder.injectDynamically(InjectorBuilder.java:175)
	at org.elasticsearch.common.inject.InjectorBuilder.build(InjectorBuilder.java:110)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:93)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:70)
	at org.elasticsearch.common.inject.ModulesBuilder.createInjector(ModulesBuilder.java:46)
	at org.elasticsearch.node.Node.<init>(Node.java:200)
	at org.elasticsearch.node.Node.<init>(Node.java:128)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:178)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:285)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
Caused by: java.lang.IllegalArgumentException: Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]
	at org.elasticsearch.index.mapper.FieldTypeLookup.checkCompatibility(FieldTypeLookup.java:117)
	at org.elasticsearch.index.mapper.MapperService.checkNewMappersCompatibility(MapperService.java:368)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:319)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:265)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:333)
	... 48 more
[2015-12-19 07:32:04,659][ERROR][gateway                  ] [Manta] failed to read local state, exiting...
java.lang.IllegalStateException: unable to upgrade the mappings for the index [evol2], reason: [Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]]
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:339)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.upgradeIndexMetaData(MetaDataIndexUpgradeService.java:116)
	at org.elasticsearch.gateway.GatewayMetaState.pre20Upgrade(GatewayMetaState.java:228)
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:87)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.elasticsearch.common.inject.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:56)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:86)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.SingleParameterInjector.inject(SingleParameterInjector.java:42)
	at org.elasticsearch.common.inject.SingleParameterInjector.getAll(SingleParameterInjector.java:66)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:85)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:201)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:880)
	at org.elasticsearch.common.inject.InjectorBuilder.loadEagerSingletons(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorBuilder.injectDynamically(InjectorBuilder.java:175)
	at org.elasticsearch.common.inject.InjectorBuilder.build(InjectorBuilder.java:110)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:93)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:70)
	at org.elasticsearch.common.inject.ModulesBuilder.createInjector(ModulesBuilder.java:46)
	at org.elasticsearch.node.Node.<init>(Node.java:200)
	at org.elasticsearch.node.Node.<init>(Node.java:128)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:178)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:285)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
Caused by: java.lang.IllegalArgumentException: Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]
	at org.elasticsearch.index.mapper.FieldTypeLookup.checkCompatibility(FieldTypeLookup.java:117)
	at org.elasticsearch.index.mapper.MapperService.checkNewMappersCompatibility(MapperService.java:368)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:319)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:265)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:333)
	... 39 more
[2015-12-19 07:32:04,808][ERROR][gateway                  ] [Manta] failed to read local state, exiting...
java.lang.IllegalStateException: unable to upgrade the mappings for the index [evol2], reason: [Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]]
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:339)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.upgradeIndexMetaData(MetaDataIndexUpgradeService.java:116)
	at org.elasticsearch.gateway.GatewayMetaState.pre20Upgrade(GatewayMetaState.java:228)
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:87)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.elasticsearch.common.inject.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:56)
	at org.elasticsearch.common.inject.ConstructorInjector.construct(ConstructorInjector.java:86)
	at org.elasticsearch.common.inject.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:104)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:47)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:887)
	at org.elasticsearch.common.inject.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:43)
	at org.elasticsearch.common.inject.Scopes$1$1.get(Scopes.java:59)
	at org.elasticsearch.common.inject.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:46)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:201)
	at org.elasticsearch.common.inject.InjectorBuilder$1.call(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorImpl.callInContext(InjectorImpl.java:880)
	at org.elasticsearch.common.inject.InjectorBuilder.loadEagerSingletons(InjectorBuilder.java:193)
	at org.elasticsearch.common.inject.InjectorBuilder.injectDynamically(InjectorBuilder.java:175)
	at org.elasticsearch.common.inject.InjectorBuilder.build(InjectorBuilder.java:110)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:93)
	at org.elasticsearch.common.inject.Guice.createInjector(Guice.java:70)
	at org.elasticsearch.common.inject.ModulesBuilder.createInjector(ModulesBuilder.java:46)
	at org.elasticsearch.node.Node.<init>(Node.java:200)
	at org.elasticsearch.node.Node.<init>(Node.java:128)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:178)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:285)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)
Caused by: java.lang.IllegalArgumentException: Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]
	at org.elasticsearch.index.mapper.FieldTypeLookup.checkCompatibility(FieldTypeLookup.java:117)
	at org.elasticsearch.index.mapper.MapperService.checkNewMappersCompatibility(MapperService.java:368)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:319)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:265)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:333)
	... 30 more
[2015-12-19 07:32:04,824][ERROR][bootstrap                ] Guice Exception: java.lang.IllegalStateException: unable to upgrade the mappings for the index [evol2], reason: [Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]]
Likely root cause: java.lang.IllegalArgumentException: Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]
	at org.elasticsearch.index.mapper.FieldTypeLookup.checkCompatibility(FieldTypeLookup.java:117)
	at org.elasticsearch.index.mapper.MapperService.checkNewMappersCompatibility(MapperService.java:368)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:319)
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:265)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.checkMappingsCompatibility(MetaDataIndexUpgradeService.java:333)
	at org.elasticsearch.cluster.metadata.MetaDataIndexUpgradeService.upgradeIndexMetaData(MetaDataIndexUpgradeService.java:116)
	at org.elasticsearch.gateway.GatewayMetaState.pre20Upgrade(GatewayMetaState.java:228)
	at org.elasticsearch.gateway.GatewayMetaState.<init>(GatewayMetaState.java:87)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at <<<guice>>>
	at org.elasticsearch.node.Node.<init>(Node.java:200)
	at org.elasticsearch.node.Node.<init>(Node.java:128)
	at org.elasticsearch.node.NodeBuilder.build(NodeBuilder.java:145)
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:178)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:285)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:35)

**Comment by dadoonet** [comment]
@jmreymond could you open a new issue about this?

**Comment by jpountz** [comment]
@jmreymond @dadoonet This is a different issue actually:

```
unable to upgrade the mappings for the index [evol2], reason: [Mapper for [docsfile] conflicts with existing mapping in other types:
[mapper [docsfile.content] cannot be changed from type [string] to [attachment]]]
```

The `evol2` index has at least two types, one of them is declaring `docsfile.content` as a `string` field and another type is declaring it as an `attachment` field. This kind of configuration caused bad bugs in elasticsearch 1.x, which is why it is not supported  anymore as of version 2.0 and the only fix is to reindex your data in different indices.

**Comment by maksutspahi** [comment]
Hello, I could not figure out how to make this work with java client api in 2.1.1 when using local node which I am using for integration tests. 
My settings:
```
Settings settings = Settings.put("plugin.types", MapperMurmur3Plugin.class.getName())
```
With the local node:
```
Node node = nodeBuilder()
                .settings(settings)
                .local(true).node();
```

While index creation it gives me :
```
Caused by: org.elasticsearch.index.mapper.MapperParsingException: no handler for type [murmur3] declared on field [hash]
	at org.elasticsearch.index.mapper.core.TypeParsers.parseMultiField(TypeParsers.java:362) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.core.StringFieldMapper$TypeParser.parse(StringFieldMapper.java:203) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:310) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:223) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.object.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:139) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:140) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.DocumentMapperParser.parseCompressed(DocumentMapperParser.java:121) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:391) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:265) ~[elasticsearch-2.1.1.jar:2.1.1]
	at org.elasticsearch.cluster.metadata.MetaDataCreateIndexService$2.execute(MetaDataCreateIndexService.java:378) ~[elasticsearch-2.1.1.jar:2.1.1]
```
My mapping is same as I posted in the first comment.

This error does not happen in 2.0.0 client. Is it related with this issue or am I not registering the plugin in the correct way ?

**Comment by dadoonet** [comment]
@maksutspahi You can not start a plugin like this from 2.1.
Read also this: https://discuss.elastic.co/t/es-plugin-add-to-node-server-programmatically/37616/5

**Comment by maksutspahi** [comment]
Thank you that worked. Also this bug seems fixed in 2.1.1 thanks.
